src/core/net/mdns.hpp

```diff
@@ -552,6 +563,71 @@ class Core : public InstanceLocator, private NonCopyable
      */
     void SetMaxMessageSize(uint16_t aMaxSize) { mMaxMessageSize = aMaxSize; }
 
+    /**
+     * Allocates a new iterator.
+     *
+     * @returns A pointer to the newly allocated iterator or `nullptr` if fails to allocate.
+     *
+     */
+    Iterator *AllocateIterator(void);
+
+    /**
+     * Frees a previously allocated iterator.
+     *
+     * @param[in] aIterator  The iterator to free.
+     *
+     */
+    void FreeIterator(Iterator &aIterator);
+
+    /**
+     * Iterates over registered host entries.
+     *
+     * On success, @p aHost is populated with the next host information. Pointers within the `Host` structure
+     * (like `mName`) remain valid until the next call to any OpenThread stack public or platform API/callback.
+     *
+     * @param[in]  aIterator   The iterator to use.
+     * @param[out] aHost       A `Host` to return the information about next host entry.
```

Comment:
```suggestion
     * @param[out] aHost       A `Host` to return information about the next host entry.
```

Reply:
Not sure adding `call` term here would make sense. This is not a function/method pointer (not a callable object).

It is an output paramter `@param[out]`, i.e. a reference to a variable to be populated by the method to return a value.

Reply:
@abtink thanks. Should we .say .... 'Host' parameter.... though?